presto-bytecode/src/main/java/com/facebook/presto/bytecode/ClassDefinition.java

```diff
@@ -266,6 +271,10 @@ public MethodDefinition declareMethod(
             ParameterizedType returnType,
             Iterable<Parameter> parameters)
     {
+        if (Iterables.size(parameters) > MAX_ALLOWED_VARARG_ARGUMENTS) {
+            throw new PrestoException(TOO_MANY_VARARG_ARGUMENTS,
+                    String.format("Number of arguments (%d) for vararg functions exceeds limit (%d)", Iterables.size(parameters), MAX_ALLOWED_VARARG_ARGUMENTS));
```

Comment:
static import `format`.
`format("Number of arguments for method %s exceeds limit %s", name, size)`.
Also the complexity of Iterables::size could be `O(n)` in the worst case depending on the underlying implementation of Iterable. Let's store it in a variable.